tests/credential_plugins_test.py

```diff
@@ -131,6 +132,76 @@ def test_hashivault_handle_auth_not_enough_args():
         hashivault.handle_auth()
 
 
+def test_aws_assumerole_with_accesssecret(monkeypatch):
+    """Test that the aws_assumerole_backend function call returns a token given
+    the access_key and secret_key."""
+    kwargs = {
+        'access_key': 'my_access_key',
+        'secret_key': 'my_secret_key',
+        'role_arn': 'the_arn',
+        'identifier': 'access_token',
+    }
+
+    def mock_getcreds(access_key, secret_key, role_arn, session_token):
+        return {
+            'access_key': 'the_access_key',
+            'secret_key': 'the_secret_key',
+            'access_token': 'the_access_token',
+            'Expiration': datetime.datetime.today() + datetime.timedelta(days=1),
+        }
+
+    monkeypatch.setattr(
+        aws_assumerole,
+        'aws_assumerole_getcreds',
+        mock_getcreds,
+    )
+
+    token = aws_assumerole.aws_assumerole_backend(**kwargs)
+    assert token == 'the_access_token'
+
+    kwargs['identifier'] = 'secret_key'
+    token = aws_assumerole.aws_assumerole_backend(**kwargs)
+    assert token == 'the_secret_key'
+
+    kwargs['identifier'] = 'access_key'
+    token = aws_assumerole.aws_assumerole_backend(**kwargs)
+    assert token == 'the_access_key'
```

Comment:
These are 3 separate tests that should show up separately in the pytest's test report.
This could be implemented by `pytest-subtest` but a better way to do this is using `@pytest.mark.parametrize`. Please move these into params.

Reply:
done